### PR TITLE
TMI2-512-code-clean-up

### PR DESCRIPTION
**Pull request by IlyasBaqqari-CabinetOffice**
## Description

[TMI2-512](https://technologyprogramme.atlassian.net/browse/TMI2-512?atlOrigin=eyJpIjoiZDBkMmJhODQ3NWI0NGNhOWIzOTg1MWYwZTM0ZDMwOTAiLCJwIjoiaiJ9) 

As per the ticket, I have went through the code making improvements where possible.

The Project Error Analysis feature of IntelliJ and ESLint collectively flagged hundreds of warnings, however a large portion of these were false positives. I have narrowed down the genuine warnings wherever I can, implementing fixes such as:

- Removing unnecessary variables and dependencies
- Fixing typos
- Formatting improvements
- Other general improvements to the code

## Type of change

Please check the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] This change requires a documentation update.

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes:

- [x] Unit Test

- [x] Integration Test (if applicable)

- [ ] End to End Test (if applicable)

## Screenshots (if appropriate):

Please attach screenshots of the change if it is a UI change:

# Checklist:

- [ ] If I have listed dependencies above, I have ensured that they are present in the target branch.
- [x] I have performed a self-review of my code.
- [ ] I have commented my code in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation where applicable.
- [ ] I have ran cypress tests and they all pass locally.

[TMI2-512]: https://technologyprogramme.atlassian.net/browse/TMI2-512?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: ormconfig.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
    "host": "localhost",
    "port": 5432,
    "username": "postgres",
-   "password": "mysecretpassword",
+   "password": "postgres",
```
Comment (jimpurvisTCO) [inline ormconfig.json:6]: I think this may be a local value you've checked in

File: src/subscription/subscription.service.ts [hunk 2]
```diff
@@ -49,18 +49,18 @@ export class SubscriptionService {
     }
 
     async findAll(): Promise<Subscription[]> {
-        const subscripionsResult = await this.subscriptionRepository.find();
-        return subscripionsResult;
+        const subscriptionsResult = await this.subscriptionRepository.find();
+        return subscriptionsResult;
```
Comment (jimpurvisTCO) [inline src/subscription/subscription.service.ts:53]: Could this be simplified down to `return await await this.subscriptionRepository.find()` ?

File: src/app.module.ts [hunk 3]
```diff
@@ -42,7 +43,7 @@ import { Unsubscribe } from './notifications/v2/unsubscribe/unsubscribe.entity';
                         SavedSearchNotification,
                     ],
                     synchronize: false,
-                    ssl: process.env.DATABASE_SSL === 'true' ? true : false,
+                    ssl: process.env.DATABASE_SSL === 'true',
```
Comment (jimpurvisTCO) [inline src/app.module.ts:46]: Was this flagged up by analysis? I believe it's mapping an env string to a boolean

File: ormconfig.json (hunk 1 above)
Comment (IlyasBaqqari-CabinetOffice) [reply ormconfig.json:6]: Good spot! will revert this.

File: src/app.module.ts (hunk 3 above)
Comment (IlyasBaqqari-CabinetOffice) [reply src/app.module.ts:46]: Analysis picked this up as a way to simplify the statement. It checks the equality and assigns the same value the same way that the ternary would have I think.

File: src/subscription/subscription.service.ts (hunk 2 above)
Comment (IlyasBaqqari-CabinetOffice) [reply src/subscription/subscription.service.ts:53]: Good call, will change it to do that!

**Review by jimpurvisTCO — APPROVED** [review]

**Pull request merged**